encryptor/dbDataCoder.go

```diff
@@ -172,9 +169,20 @@ func (*PostgresqlDBDataCoder) Encode(expr sqlparser.Expr, data []byte) ([]byte,
 				return data, nil
 			}
 			// otherwise change type and pass it below for hex encoding
-			val.Type = sqlparser.StrVal
+			val.Type = sqlparser.PgEscapeString
 			fallthrough
-		case sqlparser.PgEscapeString, sqlparser.StrVal:
+		case sqlparser.StrVal:
+			// if it is valid string, return as is. otherwise there is binary data, and we should encode it
+			if utils.IsPrintablePostgresqlString(data) {
+				return data, nil
+			}
+			val.Type = sqlparser.PgEscapeString
+			fallthrough
+		case sqlparser.PgEscapeString:
+			// valid strings we pass as is without extra encoding
+			if utils.IsPrintablePostgresqlString(data) {
+				return data, nil
+			}
 			return PgEncodeToHexString(data), nil
```

Comment:
they are not the same. in case with `StrVal` we check that it is printable string. If so, we return it as string literal as is. If it came as `select 'some string'`, then we return `'some string'`. Otherwise we change type to `PgEscapeString` that will return as `E'some string'`. We return as PgEscapeString because with such format we can encode not printable characters as hex or oct literals.

When we fall below - yes, we check second time with IsPrintable. But it will fail and encode string. 
Or we can return PgEncodeToHexString in case with StrVal too. But I prefer to change type and use fallthrough for a future if case with PgEscapeString will process it in more complicated or different way. Then we should not to maintain copy-paste